From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 9 Dec 2017 12:00:09 +0100
Subject: [PATCH] {BUG in 4.15-rc2} objtool: do not mix op_dest_type and
 op_src_type enums

gcc reports:

    arch/x86/decode.c:141:20: error: implicit conversion from
    enumeration type 'enum op_src_type' to different enumeration type
    'enum op_dest_type' [-Werror,-Wenum-conversion].
                        op->dest.type = OP_SRC_REG;
                                      ~ ^~~~~~~~~~
---
 tools/objtool/arch/x86/decode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/objtool/arch/x86/decode.c b/tools/objtool/arch/x86/decode.c
index 8acfc47af70e..540a209b78ab 100644
--- a/tools/objtool/arch/x86/decode.c
+++ b/tools/objtool/arch/x86/decode.c
@@ -138,7 +138,7 @@ int arch_decode_instruction(struct elf *elf, struct section *sec,
 			*type = INSN_STACK;
 			op->src.type = OP_SRC_ADD;
 			op->src.reg = op_to_cfi_reg[modrm_reg][rex_r];
-			op->dest.type = OP_SRC_REG;
+			op->dest.type = OP_DEST_REG;
 			op->dest.reg = CFI_SP;
 		}
 		break;
-- 
